### PR TITLE
TST #28981 comparison operation for interval dtypes

### LINKED ISSUE #28981: Possible regression in comparison operation for interval dtypes
It seems that this comparison is now failing on master when it was working in 0.25.1.  Need to look a bit more, but I don't think it's specific to this operation.

```python
import pandas as pd

s = pd.Series([pd.Interval(0, 1), pd.Interval(1, 2)], dtype="interval")  
s == "a"
```

#### 0.25.1

```
0    False
1    False
dtype: bool
```

#### master

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-3a654234a428> in <module>
----> 1 s == "a"

~/pandas/pandas/core/ops/__init__.py in wrapper(self, other)
    527         rvalues = extract_array(other, extract_numpy=True)
    528 
--> 529         res_values = comparison_op(lvalues, rvalues, op)
    530 
    531         return _construct_result(self, res_values, index=self.index, name=res_name)

~/pandas/pandas/core/ops/array_ops.py in comparison_op(left, right, op)
    253 
    254     if should_extension_dispatch(lvalues, rvalues):
--> 255         res_values = dispatch_to_extension_op(op, lvalues, rvalues)
    256 
    257     elif is_scalar(rvalues) and isna(rvalues):

~/pandas/pandas/core/ops/dispatch.py in dispatch_to_extension_op(op, left, right, keep_null_freq)
    124     #  a Series or Index.
    125 
--> 126     if left.dtype.kind in "mM" and isinstance(left, np.ndarray):
    127         # We need to cast datetime64 and timedelta64 ndarrays to
    128         #  DatetimeArray/TimedeltaArray.  But we avoid wrapping others in

TypeError: 'in <string>' requires string as left operand, not NoneType
```

@jbrockmendel Is the fix for this as simple as (say) setting the `kind` to `"O"` here https://github.com/pandas-dev/pandas/blob/master/pandas/core/dtypes/dtypes.py#L975?  It looks like we're assuming the `dtype` has a `kind` attribute when it doesn't.

### PR DESCRIPTION
**Pull request by OlivierLuG**
As a complement of #28980, one test was modified to take into account this issue:
- [ x ] closes #28981
- [ 1 ] test modified
- [ x ] passes `black pandas`
- [ x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/arithmetic/test_period.py [hunk 1]
```diff
@@ -1525,8 +1525,11 @@ def test_pi_sub_period_nat(self):
 
     @pytest.mark.parametrize("scalars", ["a", False, 1, 1.0, None])
     def test_comparison_operations(self, scalars):
-        # GH 28980
+        # GH #28980 and GH #28981
         expected = Series([False, False])
         s = Series([pd.Period("2019"), pd.Period("2020")], dtype="period[A-DEC]")
         result = s == scalars
         tm.assert_series_equal(result, expected)
+        s = pd.Series([pd.Interval(0, 1), pd.Interval(1, 2)], dtype="interval")
```
Comment (dsaxton) [inline pandas/tests/arithmetic/test_period.py:1533]: Can you put this test in /tests/arithmetic/test_interval.py?

Comment (OlivierLuG) [reply pandas/tests/arithmetic/test_period.py]: ok done, a new PR was pushed
[ x ] tests passed
[ x ] black pandas

File: pandas/tests/arithmetic/test_period.py [hunk 2]
```diff
@@ -1525,7 +1525,7 @@ def test_pi_sub_period_nat(self):
 
     @pytest.mark.parametrize("scalars", ["a", False, 1, 1.0, None])
     def test_comparison_operations(self, scalars):
-        # GH 28980
+        # GH #28980
```
Comment (dsaxton) [inline pandas/tests/arithmetic/test_period.py:1528]: Probably better not to touch the git blame here

**Comment by jreback** [comment]
thanks @OlivierLuG

**Pull request merged**